tools/webhook-router/Azure.Sdk.Tools.WebhookRouter/Routing/Router.cs

```diff
@@ -62,13 +62,17 @@ private async Task<string> GetSecretAsync(string secretName)
         {
             var secretCacheKey = $"{secretName}_secretCacheKey";
 
+            logger.LogInformation("Fetching secret with cache key: {secretCacheKey}", secretCacheKey);
+
             var cachedSecret = await cache.GetOrCreateAsync<string>(secretCacheKey, async (entry) =>
             {
+                logger.LogInformation("Cache empty, fetching secret from KeyVault for cache key: {secretCacheKey}", secretCacheKey);
+
                 entry.AbsoluteExpirationRelativeToNow = TimeSpan.FromSeconds(60);
 
                 var client = GetSecretClient();
-                var response = await client.GetSecretAsync(secretName);
-                var secret = response.Value.Value; // Urgh!
+                KeyVaultSecret response = await client.GetSecretAsync(secretName);
```

Comment:
This is why I'd normally rather see the actual class than a var. I agree with your previous assessment of response.Value.Value which was Urgh!

Reply:
Var is only problematic here because cause of the sneaky implicit cast on this type. As you can probably tell I'm in team var :)